kernel/mempool.c

```diff
@@ -56,9 +56,12 @@ void k_free(void *ptr)
 	}
 }
 
-#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+#define HEAP_MEM_POOL_SIZE MAX(CONFIG_HEAP_MEM_POOL_SIZE, \
```

Comment:
I appreciate the fix, we've also run into that issue recently.

Personally I think it would be better to just fail here if current size < min size, I know that that would mean we need to modify multiple samples, however with this approach this can be confusing to the users (i.e. they're getting a different size than they're setting).

Reply:
The question is, how exactly would we modify the applications? You could have them set the heap to the "worst case" scenario, but then that would result in something that doesn't build on boards with small amounts of RAM. You could do board-specific overlays, but that doesn't cover subsystems or SoCs that have specific requirements, and it'd also multiply the extent of changes (you'd need the same overlays for every single app that currently sets a heap size in their prj.conf).

I agree that it's not ideal that the rounding up happens silently. I was considering using something like `#warning HEAP_MEM_POOL_SIZE is less than HEAP_MEM_POOL_MIN_SIZE` but wouldn't that potentially also fail the build when `-Werror` is used?

Reply:
I think cmake `message(WARNING ....)` should be enough

Reply:
a warning sounds good

Reply:
Yep, I tried it and seems to work fine. I get this in the build output now:
```
CMake Warning at /Users/jhedberg/src/zephyr/zephyr/kernel/CMakeLists.txt:122 (message):


      CONFIG_HEAP_MEM_POOL_SIZE is less than CONFIG_HEAP_MEM_POOL_MIN_SIZE.
      The size of the system heap will be rounded up to the minimum size.
```

I've now pushed an update with this change.